components/esp_netif/include/esp_netif.h

```diff
@@ -119,6 +119,15 @@ esp_err_t esp_netif_set_driver_config(esp_netif_t *esp_netif,
  */
 esp_err_t esp_netif_attach(esp_netif_t *esp_netif, esp_netif_iodriver_handle driver_handle);
 
+/**
+ * @brief This function sets default netif no matter which implementation used
+ *
+ * @param esp_netif handle to network interface
+ *
+ * @note: This function must be called from lwip thread
+ */
+void esp_netif_set_default_netif(esp_netif_t *esp_netif);
```

Comment:
this needs to be public as if you have SLIP and WIFI, and the WIFI connection drops, slip becomes the default interface, causing any non-slip network stuff to fail for the rest of time.

in some cases this could be mitigated by `bind`ing on the correct interface, but, this option is not always exposed (for example, with the MQTT client), and, you may wish for either ethernet or wifi to work (or to prioritise one or the other) which is not easily achieved with `bind`.

exporting this allows the slip + wifi case to be resolved with something like the following on `IP_EVENT_STA_GOT_IP` or `IP_EVENT_ETH_GOT_IP` events:
```c
ip_event_got_ip_t* ip_event = (ip_event_got_ip_t*) event_data;
esp_netif_set_default_netif(ip_event->esp_netif);
```

Reply:
We have to be careful if we want to make this function public, as you can see from the note it needs to be called in lwIP context. We would need the api version of that function (similar to e.g. setup/down, start/stop -- call using `esp_netif_lwip_ipc_call()`

Understand the point, it make sense to control the routing from user code. Currently the routing priorities are assigned to every interface and the one with the highest number is chosen to be the default one. Default priorities are set here: https://github.com/espressif/esp-idf/blob/master/components/esp_netif/include/esp_netif_defaults.h#L73 and could be easily changed from user code (see that I've set the PPP  prio too high, will fix).

Would suggest lowering the default `routeprio` of the SLIP interface first. I can make setting the default interface more explicit/public API in some future update.

Reply:
ahh yeah dropping the defaults should do it, and the STA/ETH priority is the solution to another bug i've been dealing with, will revert and fix priorities / put a note in, thanks!